metadata-ingestion/src/datahub/ingestion/source/unity/connection_test.py

```diff
@@ -9,14 +12,48 @@
 from datahub.ingestion.source.unity.proxy import UnityCatalogApiProxy
 from datahub.ingestion.source.unity.report import UnityCatalogReport
 
+try:
+    from azure.identity import DefaultAzureCredential
+
+    AZURE_AVAILABLE = True
+except ImportError:
+    AZURE_AVAILABLE = False
+
+try:
+    from databricks import sql  # type: ignore
+
+    DATABRICKS_SQL_AVAILABLE = True
+except ImportError:
+    DATABRICKS_SQL_AVAILABLE = False
+
+logger = logging.getLogger(__name__)
+
+# Databricks resource ID for Azure
+DATABRICKS_RESOURCE_ID = "2ff814a6-3304-4ab8-85cb-cd0e6f879c1d"
+
 
 class UnityCatalogConnectionTest:
     def __init__(self, config: UnityCatalogSourceConfig):
         self.config = config
         self.report = UnityCatalogReport()
+
+        # Get the appropriate token based on auth type
+        if config.auth_type == "azure_default":
+            if not AZURE_AVAILABLE:
+                raise ImportError(
+                    "azure-identity package is required for Azure Default Credentials authentication. "
+                    "Please install it with: pip install azure-identity"
+                )
+            credential = DefaultAzureCredential()
+            token = credential.get_token(f"{DATABRICKS_RESOURCE_ID}/.default").token
```

Comment:
this code is duplicated across a bunch of places - refactor to reduce code duplication

I'd recommend adding auth-related things as methods on the connection config type, similar to how we do it for snowflake right now